ExampleApp/ExampleApp/Screens/Characteristics/ViewController/CharacterisiticsViewController.swift

```diff
@@ -0,0 +1,175 @@
+    }
+
+    private func addReadActions(to actionSheet: UIAlertController) {
+        let readValueNotificationAction = UIAlertAction(title: Constant.Strings.titleRead,
+                style: .default) { [weak self] _ in
+            self?.viewModel.triggerValueRead()
+        }
+        actionSheet.addAction(readValueNotificationAction)
+    }
+
+    private func addNotificationActions(to actionSheet: UIAlertController) {
+        let turnNotificationOffAction = UIAlertAction(title: Constant.Strings.turnOffNotifications,
+                style: .default) { _ in
+            self.viewModel.setNotificationsState(enabled: false)
+        }
+        let turnNotificationOnAction = UIAlertAction(title: Constant.Strings.turnOnNotifications,
+                style: .default) { _ in
+            self.viewModel.setNotificationsState(enabled: true)
+        }
+        actionSheet.addAction(turnNotificationOffAction)
+        actionSheet.addAction(turnNotificationOnAction)
+    }
+
+    private func addDismissAction(to actionSheet: UIAlertController) {
+        let dismissAction = UIAlertAction(title: Constant.Strings.titleCancel,
+                style: .cancel) { [unowned self] _ in
+            self.dismiss(animated: true, completion: nil)
+        }
+        actionSheet.addAction(dismissAction)
+    }
+
+    fileprivate func showWriteFieldForCharacteristic() {
+        let valueWriteController = UIAlertController(title: Constant.Strings.titleWriteValue,
+                message: Constant.Strings.hexValue,
+                preferredStyle: .alert)
+        valueWriteController.addTextField { _ in
+        }
+
+        valueWriteController.addAction(UIAlertAction(title: Constant.Strings.titleCancel, style: .cancel, handler: nil))
+        valueWriteController.addAction(UIAlertAction(title: Constant.Strings.titleWrite, style: .default) { _ in
+
+            if let _text = valueWriteController.textFields?.first?.text {
+                self.viewModel.writeValueForCharacteristic(hexadecimalString: _text)
+            }
+
+        })
+        present(valueWriteController, animated: true, completion: nil)
+
+    }
+
+    private func showAlert(title title: String, message: String) {
+        let alertController = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        let action = UIAlertAction(title: Constant.Strings.titleOk, style: .default) { [weak self] _ in
+            self?.dismiss(animated: true)
+        }
+        alertController.addAction(action)
+        present(alertController, animated: true)
+    }
+}
+
+extension CharacteristicsViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 140
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let characteristic = dataSource.takeItemAt(index: indexPath.row) as? Characteristic else {
+            return
+        }
```

Comment:
No need for casting it signature is typed! :)

Reply:
Done :)